boa/src/builtins/object/mod.rs

```diff
@@ -104,6 +114,100 @@ impl Object {
         }
     }
 
+    /// `Object.getOwnPropertyDescriptor( object, property )`
+    ///
+    /// Returns an object describing the configuration of a specific property on a given object.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://www.ecma-international.org/ecma-262/10.0/index.html#sec-object.getownpropertydescriptor
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertyDescriptor
+    pub fn get_own_property_descriptor(
+        _: &Value,
+        args: &[Value],
+        ctx: &mut Context,
+    ) -> Result<Value> {
+        let object = args.get(0).unwrap().to_object(ctx)?;
+        if let Some(key) = args.get(1) {
+            let key = key.to_property_key(ctx)?;
+            let desc = object.borrow().get_own_property(&key);
+            Self::from_property_descriptor(desc, ctx)
+        } else {
+            Ok(Value::undefined())
+        }
+    }
+
+    /// `Object.getOwnPropertyDescriptors( object )`
+    ///
+    /// Returns all own property descriptors of a given object.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://www.ecma-international.org/ecma-262/10.0/index.html#sec-object.getownpropertydescriptor
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertyDescriptors
+    pub fn get_own_property_descriptors(
+        _: &Value,
+        args: &[Value],
+        ctx: &mut Context,
+    ) -> Result<Value> {
+        let object = args.get(0).unwrap_or(&Value::undefined()).to_object(ctx)?;
+        let descriptors = ctx.construct_object();
+
+        for (key, _) in object.borrow().string_properties() {
```

Comment:
```suggestion
        for (key, _) in object.borrow().keys() {
```
It should iterate through all the keys not just string ones